src/HLint.hs

```diff
@@ -184,8 +186,8 @@ getIdeas cmd@CmdMain{..} settings = do
 
 -- #746: run refactor even if no hint, which ensures consistent output
 -- whether there are hints or not.
-handleRefactoring :: [Idea] -> [String] -> Cmd -> IO ()
-handleRefactoring ideas files cmd@CmdMain{..} =
+handleRefactoring :: [Idea] -> [String] -> Cmd -> [Extension] -> [Extension] -> IO ()
+handleRefactoring ideas files cmd@CmdMain{..} ys ns =
```

Comment:
Why not have handleRefactoring just do:

```
let ParseFlags{enabledExtensions, disabledExtensions} = cmdParseFlags cmd
```

Everything else just passes around the `CmdMain` flags and generates the parse flags from it when they need it. I appreciate that results in calling `cmdParseFlags` twice (more expensive), but that seems better than relying on the invariant that the flags match the command.